### PR TITLE
V-Ray CPU render fidelity tests.

### LINKED ISSUE #4483: Add V-Ray CPU to the render fidelity test
### Description

We should add V-Ray to the model-viewer render fidelity page.  This will encourage improved interoperability of the Khronos PBR BDSF with the V-Ray ecosystem.

There is an existing Python project for rendering glTFs in V-Ray here: https://github.com/ChaosGroup/vray_gltf. It is a couple years old so I believe it doesn't support the latest extensions, but maybe we can start with it.

We should likely take this existing python project and modify it to load the relevant models, cameras and environments used by the model-viewer fidelity test framework, and then submit the updated code and relevant images to this repository, allowing us to see what works and what doesn't work in V-Ray.

This can then drive further updates to the V-Ray test script (probably adding support for modern glTF extensions) or to this integration test suite.

I am willing to fund this on behalf of Threekit.

#### Live Demo

Not relevant.

### Version

Not relevant.

### Browser Affected

Not relevant.

### OS

Not relevant.

### AR

Not relevant.

### PR DESCRIPTION
**Pull request by jasondavies**
Fixes #4483.

### CODE REVIEW

**Comment by jasondavies** [comment]
Unexpectedly ran into an unknown error when leaving this to run update-screenshots:

```
vray: Rendering khronos-Cube...Error: Offline rendering process for V-Ray failed:
    at updateScreenshots (file:///home/jason/model-viewer/packages/render-fidelity-tools/lib/workflows/update-screenshots.js:130:31)
📋 Screenshot updates concluded
npm ERR! Lifecycle script `update-screenshots` failed with error:
npm ERR! Error: command failed
npm ERR!   in workspace: @google/model-viewer-render-fidelity-tools@0.0.1
npm ERR!   at location: /home/jason/model-viewer/packages/render-fidelity-tools
```

It's a shame there wasn't more information provided.

**Comment by jasondavies** [comment]
Unclear exactly why I encountered two seemingly random failures during V-Ray execution, with no errors. Not sure if error logging could be improved (it seems fine on the model-viewer side at first glance).

**Comment by jasondavies** [comment]
OK, looks like the reason for the two failures was recorded in my syslog (redacted):

```
Sep 29 21:48:58 XXX kernel: [ 8400.248403] python[11779]: segfault at 0 ip XXX sp XXX error 4 in libcgauth.so[XXX]
```

Seems like a V-Ray bug.  I'm using `appsdk-linux_qt5-release-vray_v6.10.07-20230426`.

**Comment by jasondavies** [comment]
I've added ACES filmic tone mapping now (applied to the raw image which should be linear).

vray-golden

**Comment by jasondavies** [comment]
Looks like a width/height swap bug in my array handling code.  Re-rendering goldens now.

**Comment by jasondavies** [comment]
Camera issues fixed.

**Comment by jasondavies** [comment]
Looks like the following scenarios have obvious issues:

- Sponza (looks far too dark)
- Fox (totally blank; maybe not supported at all - like Stellar?)
- Anything with renderSkybox=true e.g. MetalRoughSpheres
- RecursiveSkeletons (though Filament looks the same)
- TextureLinearInterpolationTest
- BoxVertexColors
- SimpleMorph
- TextureSettingsTest
- TriangleWithoutIndices
- DirectionalLightTest

**Comment by jasondavies** [comment]
Implemented `renderSkybox` support.  I think this is ready for review.

File: packages/render-fidelity-tools/test/renderers/vray/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## Requirements
+
+- [V-Ray App SDK](https://www.chaos.com/vray/application-sdk)
+- A V-Ray license
```
Comment (elalish) [inline packages/render-fidelity-tools/test/renderers/vray/README.md:4]: Thank you! Considering none of the maintainers of model-viewer have a V-Ray license, will someone commit to updating these renders occasionally? In particular, I'd like to know who to mention when a PR comes in that adds new sample models. @bsdorra has been kind enough to keep up with this for Stellar.

File: packages/render-fidelity-tools/test/goldens/khronos-Corset/vray-golden.png
Comment (elalish) [inline packages/render-fidelity-tools/test/goldens/khronos-Corset/vray-golden.png:1]: Would you mind pasting in screenshots from the results viewer so we can see how a few of these compare to the existing renders? They're looking pretty good, but I'd like to verify overall exposure and such.

**Review by elalish — COMMENTED** [review]
Excellent - it's great to have more path tracer comparisons! Just a couple small questions.

File: packages/render-fidelity-tools/test/renderers/vray/README.md (hunk 1 above)
Comment (jasondavies) [reply packages/render-fidelity-tools/test/renderers/vray/README.md:4]: Perhaps @bhouston?

File: packages/render-fidelity-tools/test/goldens/khronos-Corset/vray-golden.png
Comment (jasondavies) [reply packages/render-fidelity-tools/test/goldens/khronos-Corset/vray-golden.png:1]: Aren't you able to run the results viewer yourself?  You don't need a license for `npm run test`.

In any case, I've temporarily put a copy of the results viewer [here](https://www.jasondavies.com/vray-fidelity/test/results-viewer.html).

Comment (elalish) [reply packages/render-fidelity-tools/test/goldens/khronos-Corset/vray-golden.png:1]: That looks great! Thanks for making it easy for me.

File: packages/render-fidelity-tools/test/renderers/vray/README.md (hunk 1 above)
Comment (elalish) [reply packages/render-fidelity-tools/test/renderers/vray/README.md:4]: I think this is ready to merge if @bhouston agrees - I just want to make sure someone can actually update these.

Comment (bhouston) [reply packages/render-fidelity-tools/test/renderers/vray/README.md:4]: Let's do it!  This is an amazing push forward for the render fidelity tests!

Comment (elalish) [reply packages/render-fidelity-tools/test/renderers/vray/README.md:4]: @bhouston I still need you to agree that you're on the hook and able to update these Vray renders going forward. Can I depend on you?

Comment (bhouston) [reply packages/render-fidelity-tools/test/renderers/vray/README.md:4]: Yes, I can commit to that for the foreseeable future - I have it working on my computer as well.  I have also let Mr V-Ray himself, @vkoylazov, know that we are doing this and he is supportive.

Comment (elalish) [reply packages/render-fidelity-tools/test/renderers/vray/README.md:4]: Excellent, thank you!

**Comment by elalish** [comment]
Thanks! Your separate commits will continue to live on in this PR, which is the easiest way to view them anyway, so no harm done.

**Pull request merged**